indico/modules/events/tracks/templates/_track_list.html

```diff
@@ -14,31 +14,44 @@
-                                       data-title='{% trans title = track.title %}Delete track "{{ title }}"{% endtrans %}'
-                                       data-method="DELETE"
-                                       data-update="#track-list-container"></a>
-                                {% else %}
-                                    <a href="#" class="i-button danger text-color icon-remove disabled"
-                                       title="{% trans %}This track cannot be deleted because it contains accepted abstracts{% endtrans %}"></a>
-                                {% endif %}
+                        <div class="i-box-buttons hide-if-locked">
+                            <div class="toolbar">
+                                <div class="group">
+                                    <a href="#" class="i-button icon-edit js-edit-item"
+                                       title="{% trans %}Edit track{% endtrans %}"
+                                       data-href="{{ url_for('tracks.edit_track', track) }}"
+                                       data-title='{% trans title=track.title %}Edit track "{{ title }}"{% endtrans %}'
+                                       data-update="#track-list-container"
+                                       data-ajax-dialog></a>
+                                </div>
+                                <div class="group">
+                                    {% if track.can_delete(session.user) %}
+                                        <a href="#" class="i-button danger text-color icon-remove js-delete-item"
+                                           title="{% trans %}Delete track{% endtrans %}"
+                                           data-href="{{ url_for('tracks.delete_track', track) }}"
+                                           data-confirm="{% trans %}Are you sure you want to delete this track?{% endtrans %}"
+                                           data-title='{% trans title = track.title %}Delete track "{{ title }}"{% endtrans %}'
+                                           data-method="DELETE"
+                                           data-update="#track-list-container"></a>
+                                    {% else %}
+                                        <a href="#" class="i-button danger text-color icon-remove disabled"
+                                           title="{% trans %}This track cannot be deleted because it contains accepted abstracts{% endtrans %}"></a>
+                                    {% endif %}
+                                </div>
                             </div>
                         </div>
-                        <span class="i-box-description">{{ track.code }}</span>
+                        <span class="i-label small">
+                            {{ track.code }}
+                        </span>
+                        {% set session = track.default_session %}
```

Comment:
that's overwriting the flask `session` which is used e.g. for `session.user`. it only doesn't break stuff because we only access the session before this line and not after it.

actually, i don't think there's a need for this assignment at all - you even use `track.default_session` again in L52..

(i'll fix it)